Blizzard.IO.RabbitMQ/NetqRabbitConfigurator.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using Blizzard.IO.RabbitMQ.Entities;
+using EasyNetQ;
+using EasyNetQ.Topology;
+
+namespace Blizzard.IO.RabbitMQ
+{
+    public class NetqRabbitConfigurator : IRabbitConfigurator
+    {
+        private readonly IBus _netqBus;
+        private readonly IDictionary<string, IExchange> _nameToNetqExchange;
+        private readonly IDictionary<string, IQueue> _nameToNetqQueue;
+
+        public NetqRabbitConfigurator(IBus netqBus)
+        {
+            _netqBus = netqBus ?? throw new ArgumentNullException(nameof(netqBus));
+            _nameToNetqExchange = new Dictionary<string, IExchange>();
+            _nameToNetqQueue = new Dictionary<string, IQueue>();
+        }
+
+        public void Configure(RabbitConfiguration configuration)
+        {
+            _nameToNetqExchange.Clear();
+            _nameToNetqQueue.Clear(); 
+            DeclareExchanges(configuration.Exchanges);
+            DeclareQueues(configuration.Queues);
+            DeclareExchangeToQueueBindings(configuration.ExchangeToQueueBindings);
+            DeclareExchangeToExchangeBindings(configuration.ExchangeToExchangeBindings);
+        }
+
+        private void DeclareExchangeToExchangeBindings(List<RabbitBinding> exchangeToExchangeBindings)
```

Comment:
Also you can change the list you are passing in all the functions to IReadOnlyCollection
```suggestion
        private void DeclareExchangeToExchangeBindings(IReadOnlyCollection<RabbitBinding> exchangeToExchangeBindings)
```

Reply:
I refactored the class and it turns out that using all of these private methods made the class very ugly, so i put every thing in the save method, it look better than before

Reply:
I think this function is much harder to read now.
Can you explain why did removed those private function?

Reply:
כ